src/component/FetchItems.js

```diff
@@ -111,8 +112,51 @@ const FetchItems = ({ token, setToken, firestore }) => {
                 }
               });
 
+              // Update state whenever filter input changes
+              const handleFilterChange = event => {
+                setFilteredInput(event.target.value.toLowerCase());
+              };
+
+              const handleClearFieldButtonClick = event => {
+                event.preventDefault();
+                setFilteredInput('');
+              };
+
+              // function filters items off arr which chars are not equal to filteredInput
+              const filterArr = arr => {
+                return arr.filter(
+                  item =>
+                    item.id.slice(0, filteredInput.length) === filteredInput,
+                );
+              };
+
+              // update value of the items arrays within the lists object to only include
+              // the item(s) that remain after running through filterArr function
+              lists.soon.items = filterArr(lists.soon.items);
+              lists.prettySoon.items = filterArr(lists.prettySoon.items);
+              lists.notSoon.items = filterArr(lists.notSoon.items);
+              lists.inactive.items = filterArr(lists.inactive.items);
+
               return (
                 <React.Fragment>
+                  <div className="listFilter">
+                    <label>
+                      Search:
+                      <input
+                        placeholder="enter item"
+                        type="text"
+                        className="filterField"
+                        onChange={handleFilterChange}
+                        value={filteredInput}
+                      ></input>
+                    </label>
+                    <button
+                      className="clearFilter"
+                      onClick={handleClearFieldButtonClick}
+                    >
+                      x
+                    </button>
```

Comment:
This button could benefit from using `aria-label` (if I'm not mistaken) to indicate to screen reader users that this `X` means that it will clear the filter.